src/librustc/hir/map/mod.rs

```diff
@@ -506,6 +507,35 @@ impl<'hir> Map<'hir> {
         &self.forest.krate.attrs
     }
 
+    pub fn visit_module_item_likes<V>(&self, module: DefId, visitor: &mut V)
+        where V: ItemLikeVisitor<'hir>
+    {
+        let node_id = self.as_local_node_id(module).unwrap();
+
+        // Read the module so we'll be re-executed if new items
+        // appear immediately under in the module. If some new item appears
+        // in some nested item in the module, we'll be re-executed due to the reads
+        // in the loops below
+        self.read(node_id);
+
+        let module = &self.forest.krate.modules[&node_id];
+
+        for id in &module.items {
+            self.read(*id);
```

Comment:
This will give a dependency on changes to items in immediate submodules through `mod` items. This isn't desired since our passes do not look at these. Any change in a module will cause passes to re-execute for the parent module too.

Reply:
Yeah, maybe we can either:
- just not visit nested module items at all (which sounds like a bit of a foot-gun), or
- add a boolean flag to the method that let's one choose explicitly whether to visit nested module items.